core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -496,6 +497,8 @@ public void removeOldTasks()
                     try {
                         DateTime endTime = taskInfo.getStats().getEndTime();
                         if (endTime != null && endTime.isBefore(oldestAllowedTask)) {
+                            // The removal here is concurrency safe with respect to any concurrent loads: the cache has no expiration,
+                            // the taskId is in the cache, so there mustn't be an ongoing load.
                             tasks.asMap().remove(taskId);
```

Comment:
would that make sense to assert that task was actually removed to follow what comment above says.

Reply:
maybe, but i don't feel comfortable adding assertions in this class. do you want to address this as a followup?